Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -852,78 +876,91 @@ void ABenchmarkGymGameModeBase::InitMemReportTimer(const FString& MemReportInter
 }
 #endif
 
-int32 ABenchmarkGymGameModeBase::GetPlayerControllerCount() const
+void ABenchmarkGymGameModeBase::ReportAuthoritativeActorCount_Implementation(const int32 ActorCountId, const FString& WorkerID, const TArray<FActorCount>& ActorCounts)
 {
-	int32 Count = 0;
-	for (FConstPlayerControllerIterator PCIt = GetWorld()->GetPlayerControllerIterator(); PCIt; ++PCIt)
+	ActorCountMap& ActorCountMap = WorkerActorCounts.FindOrAdd(WorkerID);
+	for (const FActorCount& ActorCount : ActorCounts)
+	{
+		ActorCountMap.FindOrAdd(ActorCount.ActorClass) = ActorCount.Count;
+	}
+
+	ActorCountReportedIds.FindOrAdd(WorkerID) = ActorCountId;
+	if (ActorCountReportedIds.Num() == NumWorkers)
 	{
-		if (APlayerController* PC = PCIt->Get())
+		bool bAllWorkersInSync = true;
+		for (const auto& Pair : ActorCountReportedIds)
 		{
-			if (PC->HasAuthority())
-			{
-				++Count;
-			}
-			else if (USpatialNetDriver* SpatialDriver = Cast<USpatialNetDriver>(GetNetDriver()))
+			if (Pair.Value != ActorCountReportId)
 			{
-				// During actor authority handover, there's a period where no server will believe it has authority over
-				// the Unreal actor, but will still have authority over the entity. To better minimize this period, use
-				// the spatial authority as a fallback validation.
-				Worker_EntityId EntityId = SpatialDriver->PackageMap->GetEntityIdFromObject(PC);
-				const SpatialGDK::EntityViewElement* Element = SpatialDriver->Connection->GetView().Find(EntityId);
-				if (Element != nullptr && Element->Authority.Contains(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
-				{
-					++Count;
-				}
+				bAllWorkersInSync = false;
+				break;
 			}
 		}
+
+
+		ActorCountReportedIds.Empty();
```

Comment:
This seems wrong doesn't it? Shouldn't it be done only if `bAllWorkersInSync` is true? Also perhaps it isn't necessary, as you'd write over the top of previous values anyway.